### PR TITLE
Shell

### PR DESCRIPTION
**Pull request by tkchris93**
I'm still working on polishing this up, but I would like some initial feedback.

### CODE REVIEW

File: Labs/Shell/shell.tex [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+
+\begin{table}
+\begin{tabular}{l|l} 
+Commands & Description
+\\ \hline 
+\li{clear} & Clear the terminal screen \\
+\li{cp file1 dir1} & Create a copy of \li{file1} in \li{dir1} \\
+\li{cp file1 file2} & Create a copy of file1 and name it file2 \\
+\li{du dir1} & Display disk usage within \li{dir1} [\li{-a}, \li{-h}] \\
+\li{df dir1} & Display available disk space in \li{dir1} \\
```
Comment (groutr) [inline Labs/Shell/shell.tex]: Not according to the man page for `df`.  The df utility displays stats about free space on the entire filesystem that `dir1` is a part of.

File: Labs/Shell/shell.tex [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+\li{mkdir dir1} & Create a new directory named \li{dir1} \\
```
Comment (groutr) [inline Labs/Shell/shell.tex]: another useful command `mkdir -p /path/to/new/dir` will create all the intermediate directories.

File: Labs/Shell/shell.tex [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+To this point, we have barely scratched the surface of all the functionality that UNIX has to offer. However, the tools and commands we have addressed so far provide us with a foundation of the basics. Using the \li{os} module in Python, we can call UNIX commands. By combining Python and the UNIX commands, our toolset is automatically broadened.
+
+There are two functions within the \li{os} module specifically we will use. When wanting to run a UNIX command, use \li{os.system()}. When wanting to run a UNIX command and be able to store and manipulate the output, use \li{os.popen()}. This function returns a \li{file} object that we can read and store.
+
+\begin{lstlisting}
+TODO: Finish code box
+>>> import os
+>>> os.system("ls -l")
+<output>
+>>> files = os.popen("ls -l").read().split('\n')
```
Comment (groutr) [inline Labs/Shell/shell.tex]: Read the docs:
https://docs.python.org/2/library/os.html#os.popen

**Comment by tkchris93** [comment]
Thank you for the feedback, Ryan!

File: Labs/Shell/shell.tex [hunk 4]
```diff
@@ -0,0 +1,409 @@
...
+12 assignments.txt
+ 4 pics
+total 40
+\end{lstlisting}
+%$
+
+In the previous example, we pipe the contents of \li{assignments.txt} to \li{wc -l} using \li{cat}. When working with files specifically, it is better to use \emph{redirects}. The same output from the first example above can be achieved by running the following command:
+
+\begin{lstlisting}
+$ wc -l < assignments.rxt
```
Comment (AZaitzeff) [inline Labs/Shell/shell.tex]: is that suppose to be assignments.txt

**Comment by tkchris93** [comment]
Thanks Alexander

**Pull request merged**